src/Feature.php

```diff
@@ -112,4 +112,41 @@ public function getUserEmail($user)
 
         return Auth::user()->email;
     }
+
+    public function getUserRoles($user)
+    {
+        if ($user && $user->roles) {
+            return $user->roles;
+        }
+
+        if (Auth::guest()) {
+            return false;
+        }
+
+        $user = Auth::user();
+
+        return method_exists($user, 'getRolesForFeatureFlags')
```

Comment:
Instead of checking for the existence of the method, I think I would check if the model implements the interface.

```php
if ($user instanceof FeatureFlagsUserRoles) {
    // do something
}
```

Also, I think this should be the first check, the interface method should come before trying to get the roles column directly from the user model. We should also probably check for the interface with the `$user` variable passed as a parameter AND the `Auth::user()` as well.